.github/workflows/needs-rebase.yml

```diff
@@ -0,0 +1,26 @@
+# adapted from https://github.com/eps1lon/actions-label-merge-conflict
+
+name: "Pull Request Needs Rebase?"
+
+on:
+  # So that PRs touching the same files as the push are updated
+  push:
+  # So that the `dirtyLabel` is removed if conflicts are resolved
+  # We recommend `pull_request_target` so that github secrets are available.
+  # In `pull_request` we wouldn't be able to change labels of fork PRs
+  pull_request_target:
+    # types: [synchronize]
+    types: [opened, synchronize, reopened]
+
+jobs:
+  needs-rebase:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if PR needs rebase
+        uses: eps1lon/actions-label-merge-conflict@releases/2.x
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          dirtyLabel: "needs-rebase"
+          removeOnDirtyLabel: "no conflicts"
+          commentOnDirty: "conflicts detected, rebase needed"
+          commentOnClean: "conflicts have been resolved"
```

Comment:
I think this would get a bit spammy, just adding/removing the label withou a comment is better.

Reply:
done; the point is to trigger a notification to PR author though; if the auto-label-change doesn't trigger a notification, we may need to at least re-enable `commentOnDirty`